lib/bannerFactory.js

```diff
@@ -33,21 +35,83 @@ class BannerFactory extends BaseFactory {
      * @param {String}  config.createdBy           The username of the associated user
      * @param {Boolean} [config.isActive=false]    Whether the banner is active
      * @param {String}  [config.type='info']       Type of banner (info|warn|etc)
+     * @param {String}  [config.scope='GLOBAL']    Scope of the banner (GLOBAL|PIPELINE|BUILD)
      * @memberof BannerFactory
      */
-    create(config) {
+    async create(config) {
         if (!config.type) {
             config.type = 'info';
         }
         if (!config.isActive) {
             config.isActive = false;
         }
+        if (!config.scope) {
+            config.scope = 'GLOBAL';
+            config.scopeId = null;
+        }
+        if ((config.scope === 'PIPELINE' || config.scope === 'BUILD') && !config.scopeId) {
+            throw new Error(`scopeId is required when scope is ${config.scope}`);
+        }
```

Comment:
same here. This should be taken care by the payload validation using schema in the API endpoint.
```suggestion
        
```